src/routes/collectibles/handlers.rs

```diff
@@ -30,3 +36,76 @@ pub async fn collectibles(
             .await?,
     ))
 }
+
+/// Returns paginated collectibles.
+pub async fn collectibles_paginated(
+    context: &RequestContext,
+    chain_id: &str,
+    safe_address: &str,
+    cursor: &Option<String>,
+    trusted: Option<bool>,
+    exclude_spam: Option<bool>,
+) -> ApiResult<Json<Page<ServiceCollectible>>> {
+    let info_provider = DefaultInfoProvider::new(chain_id, &context);
+    let page_metadata = cursor
+        .as_ref()
+        .map(|cursor| PageMetadata::from_cursor(cursor));
+
+    let url = core_uri!(
+        info_provider,
+        "/v2/safes/{}/collectibles/?{}&trusted={}&exclude_spam={}", /* paginated service in core is behind v2 api. */
+        safe_address,
+        page_metadata
+            .as_ref()
+            .unwrap_or(&PageMetadata::default())
+            .to_url_string(),
+        trusted.unwrap_or(false),
+        exclude_spam.unwrap_or(true)
+    )?;
+
+    let body = RequestCached::new_from_context(url, &context, ChainCache::from(chain_id))
+        .request_timeout(collectibles_request_timeout())
+        .execute()
+        .await?;
+
+    let page = serde_json::from_str::<Page<ServiceCollectible>>(&body)?;
+
+    Ok(Json(page.map_inner(|link| {
+        map_link(
+            context,
+            link,
+            chain_id.to_string(),
+            safe_address.to_string(),
+            trusted,
+            exclude_spam,
+        )
+    })))
+}
+
+fn map_link(
+    context: &RequestContext,
+    original_link: Option<String>,
+    chain_id: String,
+    safe_address: String,
+    trusted: Option<bool>,
+    exclude_spam: Option<bool>,
+) -> Option<String> {
+    original_link.as_ref().map(|link| {
+        let cursor =
+            PageMetadata::from_cursor(link.split("?").collect::<Vec<&str>>().get(1).unwrap_or(&""))
+                .to_url_string();
```

Comment:
For the context of this PR this is ok but we need to think of ways to refactor this. This logic is spread across multiple places.

Also URL parsing/manipulation should probably be `Url` which we do have in the project.

Reply:
I agree, what do you think to create an issue to improve the manage of pagination and refactoring the source?

Reply:
Yes I think we should create one. No need to specify the solution (just that it needs to be evaluated)